server/routes/api/__test__/auth.unit.test.js

```diff
@@ -2,49 +2,88 @@ const request = require('supertest');
-  });
+  describe('POST Auth/Login', () => {
+    it('will return error with invalid email', async () => {
+      const res = await request(app)
+        .post('/api/auth/login')
+        .type('form')
+        .send({ email: 'invalidEmail@test', password: 'test' }); // send post with invalid invalid email
+      expect(res.statusCode).toEqual(400);
+      expect(res.body.error.message).toBe('Incorrect email and password combination.');
+    });
+
+    it('will return error with invalid password', async () => {
+      const res = await request(app)
+        .post('/api/auth/login')
+        .type('form')
+        .send({ email: existingUserEmail, password: 'wrongPassword' }); // send post with valid/existing email and invalid password
+      expect(res.statusCode).toEqual(400);
+      expect(res.body.error.message).toBe('Incorrect email and password combination.');
+    });
 
-  it('will return error with invalid password', async () => {
-    const res = await request(app)
-      .post('/api/auth/login')
-      .type('form')
-      .send({ email: existingUserEmail, password: 'wrongPassword' }); // send post with valid/existing email and invalid password
-    expect(res.statusCode).toEqual(400);
-    expect(res.body.error.message).toBe(
-      'Incorrect email and password combination.',
-    );
+    it('will return user object on successful login', async () => {
+      const email = 'testUser@test.com';
+      const res = await request(app)
+        .post('/api/auth/login')
+        .type('form')
+        .send({ email: existingUserEmail, password: 'test' }); // send post with valid email and password
+      expect(res.statusCode).toEqual(200);
+      expect(Object.keys(res.body)).toEqual(['_id', 'email', 'createdAt', 'updatedAt', '__v']); // should return user object, without password
+      expect(res.body.email).toEqual(email); // returned email should be correct user
+    });
   });
 
-  it('will return user object on successful login', async () => {
-    const email = 'testUser@test.com';
-    const res = await request(app)
-      .post('/api/auth/login')
-      .type('form')
-      .send({ email: existingUserEmail, password: 'test' }); // send post with valid email and password
-    expect(res.statusCode).toEqual(200);
-    expect(Object.keys(res.body)).toEqual([
-      '_id',
-      'email',
-      'createdAt',
-      'updatedAt',
-      '__v',
-    ]); // should return user object, without password
-    expect(res.body.email).toEqual(email); // returned email should be correct user
+  describe('GET Auth/Reset', () => {
+    let user;
+    beforeAll(async () => {
+      await request(app)
+        .post('/api/auth/forgot')
+        .type('form')
+        .send({ email: existingUserEmail });
+      user = await User.findOne({ email: existingUserEmail }).exec();
+    });
+
+    it('will return an error with no token', async () => {
+      const res = await request(app).get('/api/auth/reset/');
+      expect(res.statusCode).toEqual(404);
```

Comment:
Let's go with a 400 when no token is sent. The user is basically not sending the correct data in this instance. That seems like a validation error to me.

Reply:
We didn't send 404, express is sending that code since there is no route with just `/reset/` - do you want it set to 400?

Reply:
If so, I believe this will need to come from a different route, `/api/auth/reset`, since we never hit the token route without a token.

Reply:
Ahh, that's right. This is fine then.